pickhardtpayments/Payment.py

```diff
@@ -0,0 +1,207 @@
+        self._receiver = receiver
+        self._total_amount = total_amount
+        self._attempts = list()
+        self._start_time = time.time()
+
+    def __str__(self):
+        return "Payment with {} attempts to deliver {} sats from {} to {}".format(len(self._attempts),
+                                                                                  self._total_amount,
+                                                                                  self._sender[-8:],
+                                                                                  self._receiver[-8:])
+
+    @property
+    def sender(self):
+        """Returns the address of the sender of the payment.
+
+        :return: sender address for the payment
+        :rtype: str
+        """
+        return self._sender
+
+    @property
+    def receiver(self):
+        """Returns the address of the receiver of the payment.
+
+        :return: receiver address for the payment
+        :rtype: str
+        """
+        return self._receiver
+
+    @property
+    def total_amount(self):
+        """Returns the amount to be sent with this payment.
+
+        :return: The total amount of sats to be delivered from source address to destination address.
+        :rtype: int
+        """
+        return self._total_amount
+
+    @property
+    def start_time(self):
+        """Returns the time when Payment object was instantiated.
+
+        :return: time in seconds from epoch to instantiation of Payment object.
+        :rtype: float
+        """
+        return self._start_time
+
+    @property
+    def end_time(self):
+        """Time when payment was finished, either by being aborted or by successful settlement
+
+        Returns the time when all Attempts in Payment did settle or fail.
+
+        :return: time in seconds from epoch to successful payment or failure of payment.
+        :rtype: float
+        """
+        return self._end_time
+
+    @end_time.setter
+    def end_time(self, timestamp):
+        """Set time when payment was finished, either by being aborted or by successful settlement
+
+        Sets end_time time in seconds from epoch. Should be called when the Payment failed or when Payment
+        settled successfully.
+
+        :param timestamp: time in seconds from epoch
+        :type timestamp: float
+        """
+        self._end_time = timestamp
+
+    @property
+    def attempts(self):
+        """Returns all onions that were built and are associated with this Payment.
+
+        :return: A list of Attempts of this payment.
+        :rtype: list[Attempt]
+        """
+        return self._attempts
+
+    def add_attempts(self, attempts: list[Attempt]):
+        """Adds Attempts (onions) that have been made to settle the Payment to the Payment object.
+
+        :param attempts: a list of attempts that belong to this Payment
+        :type: list[Attempt]
+        """
+        self._attempts.extend(attempts)
+
+    @property
+    def fee(self):
```

Comment:
not sure about the semantics? I might be interested in the `PLANNED` fees and not only in the `SETTLED` fees

Reply:
I think this would be best implemented, when the use case is clearer to me. 
Currently the method to get the fees for the whole payment are only requested at the final summary. By then all planned Attempts were either settled or failed.
However, I did add a method for `planned_fees`that would get the sum of all fees for attempts in the payment that carry the settlement status of `planned`